File: php/features.php
<?php
declare(strict_types=1);

// Checktica SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CheckticaFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CheckticaBaseFeature();
            case "test":
                return new CheckticaTestFeature();
            default:
                return new CheckticaBaseFeature();
        }
    }
}

File: py/features.py
# Checktica SDK feature factory

from feature.base_feature import CheckticaBaseFeature
from feature.test_feature import CheckticaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CheckticaBaseFeature(),
        "test": lambda: CheckticaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
